### PR TITLE
UTY-1087: Test Project for Generated Code

### PR DESCRIPTION
**Pull request by jamiebrynes7**
# PR is not as big as it looks

#### Description
Created test project for all our test schema and test generated code.
Mostly moved things around.
Some bugfixes in the GDK tools config.
Updated shell scripts
#### Tests
This is tests! 
#### Documentation
N/A
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: ci/test.sh [hunk 1]
```diff
@@ -27,7 +28,8 @@ rm "${TOOLS_TEST_RESULTS_FILES}" \
 
 markEndOfBlock "Setup variables"
 
-cleanUnity
+cleanUnity "$(pwd)/workers/unity"
```
Comment (firtina-improbable) [inline ci/test.sh]: may be nice to have pwd/workers/unity and pwd/test-project to be vars instead of inline

Comment (firtina-improbable) [reply ci/test.sh]: can also be reused for the dotnet run args and `pushd`

File: test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs [hunk 2]
```diff
@@ -31,10 +32,10 @@ public void TearDown()
             world.Dispose();
         }
 
-        protected struct SomeOtherComponent : IComponentData, ISpatialComponentData
+        protected struct SomeOtherComponent : ISpatialComponentData, IComponentData
```
Comment (firtina-improbable) [inline test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs]: 🤔

**Comment by jamiebrynes7** [comment]
@firtina-improbable Am planning to write a meta test runner to replace the shell script to make things much nicer - https://improbableio.atlassian.net/browse/UTY-1158

Please ignore minor things with it 🙏

File: test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs (hunk 2 above)
Comment (jamiebrynes7) [reply test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs]: Yeah this was weird - I'm scared that this was silently failing for ages..?

Comment (firtina-improbable) [reply test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs]: I would not expect this to fail though unless we have some interesting linting things

Comment (jamiebrynes7) [reply test-project/Assets/EditmodeTests/GameObjectRepresentation/Readers/ReaderWriterTestsBase.cs]: Wait I thought this comment was about something else - this was me running the linter over all the tests 😉

File: ci/test.sh [hunk 3]
```diff
@@ -46,37 +48,52 @@ markEndOfBlock "Code Generator Testing"
...
+        -logfile "${PROJECT_DIR}/logs/unity-editmode-test-run.log" \
+        -testResults "${EDITMODE_TEST_RESULTS_FILE}"
 
     EDITMODE_TEST_RESULT=$?
 popd
 
 markEndOfBlock "Editmode Testing"
 
-cleanUnity
+cleanUnity "$(pwd)/workers/unity"
```
Comment (jared-improbable) [inline ci/test.sh]: We really need better worker names than `unity`

File: test-project/Assets/Generated/Source/improbable/Coordinates.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// ===========
```
Comment (jared-improbable) [inline test-project/Assets/Generated/Source/improbable/Coordinates.cs]: Was this intended to be committed?

File: test-project/ProjectSettings/Physics2DSettings.asset [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  m_AlwaysShowColliders: 0
+  m_ShowColliderSleep: 1
+  m_ShowColliderContacts: 0
+  m_ShowColliderAABB: 0
+  m_ContactArrowScale: 0.2
+  m_ColliderAwakeColor: {r: 0.5686275, g: 0.95686275, b: 0.54509807, a: 0.7529412}
+  m_ColliderAsleepColor: {r: 0.5686275, g: 0.95686275, b: 0.54509807, a: 0.36078432}
+  m_ColliderContactColor: {r: 1, g: 0, b: 1, a: 0.6862745}
+  m_ColliderAABBColor: {r: 1, g: 1, b: 0, a: 0.2509804}
+  m_LayerCollisionMatrix: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```
Comment (jared-improbable) [inline test-project/ProjectSettings/Physics2DSettings.asset]: Coincidentally the sound I make while viewing Unity YAMLs

File: test-project/Assets/Scenes/SampleScene.unity [hunk 6]
```diff
@@ -0,0 +1,258 @@
+%YAML 1.1
```
Comment (jared-improbable) [inline test-project/Assets/Scenes/SampleScene.unity]: `Tests.unity`?

File: ci/test.sh (hunk 3 above)
Comment (jamiebrynes7) [reply ci/test.sh]: Probably yeah - not sure what we want for this

File: test-project/Assets/Generated/Source/improbable/Coordinates.cs (hunk 4 above)
Comment (jamiebrynes7) [reply test-project/Assets/Generated/Source/improbable/Coordinates.cs]: I must not have ignored the std lib generated stuff - not

File: test-project/Assets/Scenes/SampleScene.unity (hunk 6 above)
Comment (jamiebrynes7) [reply test-project/Assets/Scenes/SampleScene.unity]: Sure 😄. The scene isn't used for anything yet 🔜 ™️

File: .gitignore [hunk 7]
```diff
@@ -1,5 +1,6 @@
 /build
 /workers/unity/build/
+/test-project/build
```
Comment (samiwh) [inline .gitignore]: Possibly annoying comment. If this only contains codegen tests can we call it something that reflects that. codegen-test-project or something. test-project is pretty ambiguous.

Comment (jamiebrynes7) [reply .gitignore]: It is an annoying comment 😛

**Pull request merged**